src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_credentials/SshCredentialDialog.java

```diff
@@ -36,6 +42,19 @@ protected void resolve(String caption) {
      * Adds this credential and close the modal dialog.
      */
     public void add() {
-        find(by.id("credentials-add-submit-button")).click();
+        final By addSubmitButton = by.id("credentials-add-submit-button");
+
+        find(addSubmitButton).click();
+
+        waitFor().withTimeout(5, TimeUnit.SECONDS).until(new Callable<Boolean>() {
+            @Override
+            public Boolean call() throws Exception {
+                try {
+                    return !find(addSubmitButton).isDisplayed();
+                } catch (final NoSuchElementException | StaleElementReferenceException ex) {
```

Comment:
I think it is not valid for this situation. Ignoring such exceptions will always make the test fail due to timeout, as the button is not there (which is actually expected) and therefore a timeout will be thrown.

I need the exceptions to be thrown to detect the button has already been hidden and then return true.